run/service.go

```diff
@@ -82,6 +84,46 @@ func (s *Service) GetScheduledRun(ctx context.Context, namespace models.Namespac
 	return jobRun, err
 }
 
+func (s *Service) GetJobRunList(ctx context.Context, projectSpec models.ProjectSpec, jobSpec models.JobSpec, jobQuery *models.JobQuery) ([]models.JobRun, error) {
+	var jobRuns []models.JobRun
+	if jobQuery.OnlyLastRun {
+		return s.scheduler.GetJobRuns(ctx, projectSpec, jobQuery)
+	}
+	//validate job query
+	err := validateJobQuery(jobQuery, jobSpec)
+	if err != nil {
+		return jobRuns, err
+	}
+	//check for manual runs
+	jobInterval := jobSpec.Schedule.Interval
+	var expectedRuns []models.JobRun
+	if jobInterval != "" {
+		sch, err := cron.ParseCronSchedule(jobInterval)
+		if err != nil {
+			return jobRuns, fmt.Errorf("unable to parse the interval from DB %w", err)
+		}
+		//modify job query
+		jobQuery = modifyDateRange(jobQuery, jobSpec, sch)
+		//get expected runs
+		expectedRuns = getExpectedRun(sch, jobQuery.StartDate, jobQuery.EndDate)
+	} else {
+		jobQuery.IncludeManualRun = true
+	}
+	//call to airflow for get runs
+	actualRuns, err := s.scheduler.GetJobRuns(ctx, projectSpec, jobQuery)
+	if err != nil {
+		return jobRuns, fmt.Errorf("unable to get job runs from airflow %w", err)
+	}
+	//mergeRuns
+	var totalRuns []models.JobRun
+	if len(expectedRuns) > 0 {
+		totalRuns = mergeRuns(expectedRuns, actualRuns)
+	}
+	//filterRuns
+	result := filterRuns(totalRuns, createFilterSet(jobQuery.Filter))
```

Comment:
Two scenarios I believe won't work as expected.
1. for jobs with no schedule, we don't return any jobruns.
2. for users requesting manual runs, they won't see any change in behaviour, they will still get only scheduled runs.

Reply:
as discuss, removed the manual run logic